app/src/main/java/org/mozilla/focus/search/SearchEngineManager.java

```diff
@@ -339,4 +340,8 @@ public static void removeSearchEngines(Set<String> engineIdsToRemove, SharedPref
     public static void restoreDefaultSearchEngines(SharedPreferences sharedPreferences) {
         sharedPreferences.edit().remove(PREF_KEY_HIDDEN_DEFAULT_ENGINES).apply();
     }
+
+    private void invalidateSearchEngines() {
```

Comment:
I think this method needs to be `synchronized`. Otherwise another thread reading it might not see it that is has changed, e.g.:
https://www.javacodegeeks.com/2014/08/java-concurrency-tutorial-visibility-between-threads.html